### PR TITLE
gromacs: Let runtime logs show homebrew was used

### PR DESCRIPTION
**Pull request by mabraham**
Knowing that homebrew has patched the code and organized the build is
potentially valuable information for GROMACS users and developers
troubleshooting their runs.

- [x] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/HEAD/CONTRIBUTING.md)?
- [x] Have you ensured that your commits follow the [commit style guide](https://docs.brew.sh/Formula-Cookbook#commit)?
- [x] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [ ] Have you built your formula locally with `brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [ ] Is your test running fine `brew test <formula>`, where `<formula>` is the name of the formula you're submitting?
- [ ] Does your build pass `brew audit --strict <formula>` (after doing `brew install --build-from-source <formula>`)? If this is a new formula, does it pass `brew audit --new <formula>`?

-----

### CODE REVIEW

File: Formula/gromacs.rb [hunk 1]
```diff
@@ -50,7 +50,8 @@ def install
     inreplace "src/gromacs/gromacs-config.cmake.cmakein", "@GROMACS_CXX_COMPILER@", cxx
 
     mkdir "build" do
-      system "cmake", "..", *std_cmake_args, "-DGROMACS_CXX_COMPILER=#{cxx}"
+      system "cmake", "..", *std_cmake_args, "-DGROMACS_CXX_COMPILER=#{cxx}",
+                      "-DGMX_VERSION_STRING_OF_FORK=homebrew"
```
Comment (carlocab) [inline Formula/gromacs.rb:54]:
```suggestion
      system "cmake", "..", *std_cmake_args, "-DGROMACS_CXX_COMPILER=#{cxx}",
                                             "-DGMX_VERSION_STRING_OF_FORK=#{tap.user}"
```
Let's align the `-D` flags for `cmake`, and use `tap.user` to make this formula easy to reuse for external taps.

**Review by carlocab — COMMENTED** [review]
Thank, @mabraham. A few suggestions.

I noticed that your PR branch is also called `master`. Can you reopen your PR using a PR branch of a different name? @BrewTestBot has had difficulty merging PRs with branches also named `master` as of late. (Querying the GitHub API for workflows associated with a PR is broken when the PR branch is named `master`.)

When you do so, please use the commit message
```
gromacs: let runtime logs show homebrew was used
```
or similar, to conform with Homebrew/core style. See, for example, the [commit style guide](https://docs.brew.sh/Formula-Cookbook#commit):

> The preferred commit message format for simple version updates is `foobar 7.3` and for fixes is `foobar: fix flibble matrix.`.

File: Formula/gromacs.rb (hunk 1 above)
Comment (mabraham) [reply Formula/gromacs.rb:54]: Done!

**Comment by mabraham** [comment]
New PR at #85158

**Pull request closed without merging**